src/test/java/app/bpartners/geojobs/endpoint/rest/security/authorizer/CommunityFullDetectionAuthorizerIT.java

```diff
@@ -0,0 +1,230 @@
+package app.bpartners.geojobs.endpoint.rest.security.authorizer;
+
+import static app.bpartners.geojobs.endpoint.rest.security.model.Authority.Role.ROLE_ADMIN;
+import static app.bpartners.geojobs.endpoint.rest.security.model.Authority.Role.ROLE_COMMUNITY;
+import static app.bpartners.geojobs.repository.model.detection.DetectableType.PATHWAY;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mockStatic;
+
+import app.bpartners.geojobs.conf.FacadeIT;
+import app.bpartners.geojobs.endpoint.rest.model.CreateFullDetection;
+import app.bpartners.geojobs.endpoint.rest.model.DetectableObjectType;
+import app.bpartners.geojobs.endpoint.rest.model.Feature;
+import app.bpartners.geojobs.endpoint.rest.model.MultiPolygon;
+import app.bpartners.geojobs.endpoint.rest.security.AuthProvider;
+import app.bpartners.geojobs.endpoint.rest.security.model.Authority;
+import app.bpartners.geojobs.endpoint.rest.security.model.Principal;
+import app.bpartners.geojobs.model.exception.ForbiddenException;
+import app.bpartners.geojobs.repository.CommunityAuthorizationRepository;
+import app.bpartners.geojobs.repository.model.community.CommunityAuthorization;
+import app.bpartners.geojobs.repository.model.community.CommunityAuthorizedZone;
+import app.bpartners.geojobs.repository.model.community.CommunityDetectableObjectType;
+import app.bpartners.geojobs.service.CommunityUsedSurfaceService;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class CommunityFullDetectionAuthorizerIT extends FacadeIT {
+  private final MockedStatic<AuthProvider> authProvider = mockStatic(AuthProvider.class);
+  private static final String COMMUNITY_ID = "communityId";
+  private static final String COMMUNITY_API_KEY = "communityApiKey";
+  private static final String ADMIN_API_KEY = "the-admin-api-key";
+
+  @Autowired CommunityFullDetectionAuthorizer subject;
+  @Autowired CommunityAuthorizationRepository caRepository;
```

Comment:
Write unit tests for `caRepository` and `communityUsedSurfaceService`. Then, here, you can mock their behavior to ensure code readability and maintainability.